src/components/batch_select.vue

```diff
@@ -0,0 +1,188 @@
+<template>
+  <div class="batch-select">
+    <button
+      class="button white-button batch-select-button"
+      @click="d_show_batch_select_modal = true"
+    >
+      Batch Select
+    </button>
+    <div @click.stop>
+      <modal
+        v-if="d_show_batch_select_modal"
+        @close="d_show_batch_select_modal = false"
+        size="large"
+        click_outside_to_close
+      >
+        <div class="modal-header">
+          Select Items ({{ d_selected_items.length }} out of
+          {{ choices.length }} items selected)
+        </div>
+        <input
+          class="input batch-search-field"
+          type="text"
+          placeholder="Enter a name"
+          v-model="d_batch_search_query"
+        />
+        <div>
+          <ul class="batch-select-card-grid">
+            <li
+              class="batch-select-card"
+              :class="{
+                selected: d_selected_items.some((el) => el.pk === item.pk),
+              }"
+              v-for="item of batch_filtered_items"
+              :key="item.pk"
+              @click="batch_toggle_select(item)"
+            >
+              {{ item.name || item.pattern }}
+            </li>
+          </ul>
+        </div>
+        <div class="button-footer-right modal-button-footer">
+          <button
+            class="modal-confirm-button"
+            @click="confirm_selection"
+          >
+            Confirm
+          </button>
+          <button
+            class="modal-cancel-button"
+            @click="d_show_batch_select_modal = false"
+          >
+            Cancel
+          </button>
+        </div>
+      </modal>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { Component, Prop, Vue, Watch } from 'vue-property-decorator';
+
+import _ from 'lodash';
+
+import Modal from "@/components/modal.vue";
+
+// types for function props
+type Comparator = (lhs: unknown, rhs: unknown) => boolean;
+type NameGetter = (item: unknown) => string;
+
+@Component({
+  components: {
+    Modal
+  }
+})
+export default class BatchSelect extends Vue {
+  @Prop({required: true, type: Array})
+  choices!: unknown[];
+
+  @Prop({type: Array})
+  value!: unknown[];
+
+  @Prop({type: Function})
+  are_items_equal!: Comparator;
+
+  @Prop({type: Function})
+  get_item_name!: NameGetter;
```

Comment:
Since the component is agnostic of the data that is being passed in, we need a way to compare two unknowns (for `batch_toggle_select`) and to get name from them (for filtering in `batch_filtered_items`). Functions seem the way to go if we are assuming nothing about the items. 

However, if we assume that the items are objects with valid string and value pairs, then maybe we can pass in strings corresponding to the keys of the object that give us the unique id and name. For example, we can create a prop called `name` and pass in "pattern" so that we can have something like `item[this.name]` and that would work for student files.

Reply:
That's more or less the direction we'll want to go--I'll take a closer look on Monday and follow up then. Thanks!